### ISSUE TITLE
What to do with Snippets?

### ISSUE THREAD

**Issue by sypets** [TYPO3-Documentation/T3DocTeam#74]
[Snippets](https://docs.typo3.org/typo3cms/Snippets/)

## Questions

* Is this section maintained? By whom?
* Is this up-to-date?
* How are people supposed to know if it's up-to-date?
* Should snippets be clearly marked as unmaintained community section?
* If it's unmaintained, shouldn't we take it down?

## Problem

The problem I see with large parts of the documentation: It is not all maintained and up to date. And there is no way to find out as a newbie if it's up-to-date. Letting people fall into this trap shows no respect for time and effort people put into trying to learn TYPO3. I find this unacceptable. 

On the other hand, there are useful things in the Snippets. I think we should not remove it without any further announcements. This would show no respect for the effort contributors put into this.

## These questions depend on general decisions:

* Should documentation on docs.typo3.org be up-to-date and correct in general? Should we take down documentation that we cannot maintain? Should we rather have less but in higher quality?

My suggestion would be to come to a general decision for these general questions ASAP.

## My suggestion for the snippets

For the snippets, I think StackOverflow might be an alternative. Adding individuals snippets as StackOverflow questions and answers has the following advantages:

*  The questions and answers can be updated and maintained by community
* People put their name on it and their reputation depends on it. That is a better incentive for maintaining questions and answers than contributing on docs.typo3.org
* If someone disagrees, another answer can be added. You can vote for answers. You can comment them if you think they are wrong. 

## Suggestion for plan:

1. Announce to take down snippets in say half a year
2. Give people information about what they can do (e.g. migrate their snippets to StackOverflow)
3. Shutdown snippets in half a year.

**Comment by sypets** [comment]
There is a decision: https://decisions.typo3.org/t/what-to-do-with-the-snippets/591

**Comment by sypets** [comment]
Revised plan:

* [x] Remove snippets from main menu on https://docs.typo3.org
* [x] Add "no longer maintained banner" to snippets with link to more information
* [x] Archive [snippets repo](https://github.com/TYPO3-Documentation/TYPO3CMS-Snippets)
* Figure out where to put Examples, snippets, howtos (not scope of this issue) - the issue was already raised if we should have one central Howto / Examples section or if the examples should be available in the respective topic and then of course there are some examples that combine several topics. This should be done in combination with looking at the structure in general.
*  migrate the most useful snippets and examples from "Snippets"? 
* [ ] .... (some time in the future) remove snippets from rendering and redirect to start page

to recap: There are now several alternatives for snippets that are already being used or might be used in the future:

* Stack Overflow (use tag TYPO3)
* examples in the respective section of other manuals
* blog entries outside of docs.typo3.org

**Comment by sypets** [comment]
Is done.
